src/components/dialogs/ag-grid-table-rhf/custom-ag-grid-table.js

```diff
@@ -146,34 +137,42 @@ export const CustomAgGridTable = ({
             });
             remove(idx);
         });
-        setRowData(getValues(name));
     };
 
     const getIndex = (val) => {
-        return getValues(name).findIndex((row) => row.id === val.id);
+        return getValues(name).findIndex((row) => row.rowUuid === val.rowUuid);
     };
+
     const handleMoveRowUp = () => {
-        selectedRows.forEach((val, index) => {
-            const idx = getIndex(val);
-            move(idx, idx - 1);
-        });
-        setUpDisabled(selectedRows.some((r) => getIndex(r) === 0));
-        setDownDisabled(
-            selectedRows.some((r) => getIndex(r) === rowData.length - 1)
-        );
-        setRowData(getValues(name));
+        selectedRows
+            .map((row) => getIndex(row))
+            .sort()
+            .forEach((idx) => {
+                move(idx, idx - 1);
+                if (idx === 1) {
+                    setUpDisabled(true);
```

Comment:
You don't need to do that, check DndTable
https://github.com/gridsuite/gridstudy-app/blob/fe947e88ac5f254166b62481aa19692cebe6c84e/src/components/utils/dnd-table/dnd-table-bottom-right-buttons.js#L36
It's enough. No need to check at every loop iteration and no need to use a state.